src/CashuWallet.ts

```diff
@@ -863,6 +867,153 @@ class CashuWallet {
 		return states;
 	}
 
+	/**
+	 * Register a callback to be called whenever a mint quote's state changes
+	 * @param quoteIds List of mint quote IDs that should be subscribed to
+	 * @param callback Callback function that will be called whenever a mint quote state changes
+	 * @param errorCallback
+	 * @returns
+	 */
+	async onMintQuoteUpdates(
+		quoteIds: Array<string>,
+		callback: (payload: MintQuoteResponse) => void,
+		errorCallback: (e: Error) => void
+	): Promise<SubscriptionCanceller> {
+		await this.mint.connectWebSocket();
+		if (!this.mint.webSocketConnection) {
+			throw new Error('failed to establish WebSocket connection.');
+		}
+		const subId = this.mint.webSocketConnection.createSubscription(
+			{ kind: 'bolt11_mint_quote', filters: quoteIds },
+			callback,
+			errorCallback
+		);
+		return () => {
+			this.mint.webSocketConnection?.cancelSubscription(subId, callback);
+		};
+	}
+
+	/**
+	 * Register a callback to be called whenever a melt quote's state changes
+	 * @param quoteIds List of melt quote IDs that should be subscribed to
+	 * @param callback Callback function that will be called whenever a melt quote state changes
+	 * @param errorCallback
+	 * @returns
+	 */
+	async onMeltQuotePaid(
+		quoteId: string,
+		callback: (payload: MeltQuoteResponse) => void,
+		errorCallback: (e: Error) => void
+	): Promise<SubscriptionCanceller> {
+		await this.mint.connectWebSocket();
+		if (!this.mint.webSocketConnection) {
+			throw new Error('failed to establish WebSocket connection.');
+		}
+		const subId = this.mint.webSocketConnection.createSubscription(
+			{ kind: 'bolt11_melt_quote', filters: [quoteId] },
+			(p: MeltQuoteResponse) => {
+				if (p.state === MeltQuoteState.PAID) {
+					callback(p);
+				}
+			},
+			errorCallback
+		);
+		return () => {
+			this.mint.webSocketConnection?.cancelSubscription(subId, callback);
+		};
+	}
+
+	/**
+	 * Register a callback to be called when a single mint quote gets paid
+	 * @param quoteId Mint quote id that should be subscribed to
+	 * @param callback Callback function that will be called when this mint quote gets paid
+	 * @param errorCallback
+	 * @returns
+	 */
+	async onMintQuotePaid(
+		quoteId: string,
+		callback: (payload: MintQuoteResponse) => void,
+		errorCallback: (e: Error) => void
+	): Promise<SubscriptionCanceller> {
+		await this.mint.connectWebSocket();
+		if (!this.mint.webSocketConnection) {
+			throw new Error('failed to establish WebSocket connection.');
+		}
+		const subId = this.mint.webSocketConnection.createSubscription(
+			{ kind: 'bolt11_mint_quote', filters: [quoteId] },
+			(p: MintQuoteResponse) => {
+				if (p.state === MintQuoteState.PAID) {
+					callback(p);
+				}
+			},
+			errorCallback
+		);
+		return () => {
+			this.mint.webSocketConnection?.cancelSubscription(subId, callback);
+		};
+	}
```

Comment:
Can this not use `onMintQuoteUpdates`? Seems like a lot of repeated boiler plate.

Reply:
Very good point! Done